### ISSUE TITLE
Entering foreign key fields is broken

### ISSUE THREAD

**Issue by balajiramanathan** [pawelsalawa/sqlitestudio#3998]
### Details
When I enter any value in a foreign key field and tab to the next field, the value automatically changes back to 1 (in my database that is the lowest valid value for that foreign key - it may change to something different in your database depending on the contents of the table to which that foreign key field refers)

### Steps to reproduce
Edit any table with a foreign key field.  Try to update the foreign key field in an existing record or insert a new record and try to enter a valid value in the foreign key field.  It will change to something else.  The only way to get my desired value to stay in that field was to use the dropdown arrow and pick the record I wanted from the foreign key table (which can be very tedious if that table contains a lot of data) or pasting a value into that field instead of typing a value in.

### Operating system
Windows 7 Home

### SQLiteStudio version
3.3.0  SQLite version 3.34.1

**Comment by pawelsalawa** [comment]
Indeed there is a bug, but to me it behaves differently - if I try to type manually a correct value - it works. But if I enter value that is no reflected in the foreign column - then the application crashes. That certainly needs to be fixed.

From what I understand your case is different, is it? If I understood you correctly, you mean that if you type a valid value, but you do it manually, it doesn't accept it, but instead returns to other value?
If that's so, would it be possible for you to share the database here (you can atttach files to posts) where I can reproduce your behavior? If not, are you able to create just 2 tables in some fresh database file and populate it with enough values to reproduce your problem?

I'm asking for it, because I cannot really reproduce the issue that you have - for me it either behaves correctly, or simply crashes (as described above).

**Comment by balajiramanathan** [comment]
I have attached a small test database with two tables.  Not much data.

Unless I select from the dropdown, the second field in the primary table always goes to 1 on my computer.  I have to record a video to show you if it doesn't behave that way for you.  When I enter an invalid value, nothing happens - the value just gets reset to a valid value automatically.  There is no crash for me.

But there is a bigger issue for me.  When I try deleting rows from the primary table, I can't commit the changes because there is some error about the foreign keys.  I can understand if I try to delete rows from the foreign key table, there may be problem because the cascading update and delete may not be working correctly, but I should always be able to delete from the primary table without any issues.  Maybe this is a separate bug.

There seems to be no file attach capability that I can find, so I have put the file up on Google drive.  Here is the [link ](https://drive.google.com/file/d/1oVsOxP6-9JrzN2GiedvcuqmQnRV0-wa-/view?usp=sharing) to it.

**Comment by pawelsalawa** [comment]
This post is just to quickly upload the file here.
[Test.zip](https://github.com/pawelsalawa/sqlitestudio/files/6043712/Test.zip)

I will verify this a bit later.

**Comment by pawelsalawa** [comment]
The bug was related to changes to `QComboBox` in recent Qt versions. This is fixed now and behavior from SQLiteStudio 3.2.1 is restored.

Regarding the "Foreign key mismatch" error when you try to delete anything is because tables in the example you provided are set up incorrectly according to SQLite. It will work fine if you have foreign keys disabled (https://sqlite.org/pragma.html#pragma_foreign_keys) and this is still disabled by default in SQLite, but SQLiteStudio follows good practices and enables FK validation. You can still override this by executing mentioned PRAGMA, but I would rather recommend you to fix your database schema to follow proper FK references, which I explain below.

Quote from https://www.sqlite.org/foreignkeys.html#fk_indexes

> Usually, the parent key of a foreign key constraint is the primary key of the parent table. If they are not the primary key, then the parent key columns must be collectively subject to a UNIQUE constraint or have a UNIQUE index.

In other words, your "ForeignTable" must have `PRIMARY KEY` or `UNIQUE` clause on the column that you refer from your "PrimaryTable".
